### DISCUSSION TITLE
Upsert only metadata

### DISCUSSION THREAD

**Discussion by fferroni** [milvus-io/milvus#37282] [Q&A and General discussion]
Hello,

Is it possible to insert additional metadata to an existing entry in a collection (assuming that dynamic metadata is enabled)?

Let’s say we have a $meta field with “a” and “b” keys, and we want to add an additional “c” key.
Or we want to modify “b” key (upsert) but not modify the embedding / vector index?

Thank you

**Comment by yhmo** [comment]
With dynamic field enabled, you can add additional key-value with insert.
```
collection.insert(data={"id": 1, "vector": [], "a": 2, "b": 0.5})
collection.insert(data={"id": 1, "vector": [], "a": 2, "b": 0.5, "c": 100})
```

But if you want to update an existing row to change a key-value, you need to provide the entire data of this row for upsert() interface.
`collection.upsert(data={"id": 1, "vector": [], "a": 2, "b": 99})`

The Milvus v2.5.x will support "partial upsert", which allows users to modify a row without providing the entire row data.

**Comment by xiaofan-luan** [comment]
> Hello,
> 
> Is it possible to insert additional metadata to an existing entry in a collection (assuming that dynamic metadata is enabled)?
> 
> Let’s say we have a $meta field with “a” and “b” keys, and we want to add an additional “c” key. Or we want to modify “b” key (upsert) but not modify the embedding / vector index?
> 
> Thank you

Right now this is not supported. You have to upsert the whole role each time.
In Milvus 3.0 partitial upsert could be introduced, but again this is a simple wrapper to read the entire entity, modify and write back as an new entry
